src/test/java/com/example/bookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,165 @@
+import org.springframework.jdbc.datasource.init.ScriptUtils;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class BookControllerTest {
+    protected static MockMvc mockMvc;
+
+    @Autowired
+    private BookService bookService;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(@Autowired WebApplicationContext applicationContext) {
+        mockMvc = MockMvcBuilders.webAppContextSetup(applicationContext)
+                .apply(springSecurity()).build();
+
+    }
+
+    @BeforeEach
+    void beforeEach(@Autowired DataSource dataSourced) throws SQLException {
+        teardown(dataSourced);
+        try (Connection connection = dataSourced.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/add-three-books.sql"));
+        }
+    }
+
+    @SneakyThrows
+    private static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/remove-all-books.sql")
+            );
+        }
+    }
+
+    @WithMockUser(username = "admin", roles = "ADMIN")
+    @Test
+    void createBook_ValidRequestDto_Success() throws Exception {
+        // given
+        CreateBookRequestDto createBookRequestDto = TestUtil.createBookRequestDto();
+
+        BookDto expected = new BookDto()
+                .setId(4L)
+                .setAuthor(createBookRequestDto.getAuthor())
+                .setPrice(createBookRequestDto.getPrice())
+                .setTitle(createBookRequestDto.getTitle())
+                .setIsbn(createBookRequestDto.getIsbn())
+                .setDescription(createBookRequestDto.getDescription())
+                .setCategoryIds(createBookRequestDto.getCategoryIds());
+
+        String jsonRequest = objectMapper.writeValueAsString(createBookRequestDto);
+
+        // when
+        MvcResult result = mockMvc.perform(post("/books")
+                        .content(jsonRequest)
+                        .contentType(org.springframework.http.MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isCreated())
+                .andReturn();
+        // then
+        BookDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(),
+                BookDto.class);
+        reflectionEquals(expected, actual,"id");
```

Comment:
this method returns a boolean value. You should use it with `assertTrue`. Fix it everywhere